core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -183,4 +182,53 @@ private void calcAscendDescend(final PathWrapper rsp, final PointList pointList)
         rsp.setAscend(ascendMeters);
         rsp.setDescend(descendMeters);
     }
-}
\ No newline at end of file
+
+    private String unconnectedListString(List<Path> paths) {
```

Comment:
Could we move these two methods into the ConnectionNotFoundException class to hide that logic? This would help to keep the PathMerger class a bit slimmer.

So we might have a new Constructor for this class accepting a List<Path> and a PointList?